lib/tools/getYouTubeRevenue.ts

```diff
@@ -0,0 +1,202 @@
+        tokenValidation.tokens!.refresh_token ?? undefined
+      );
+      
+      const channelResponse = await youtube.channels.list({
+        part: ["id"],
+        mine: true,
+      });
+
+      if (!channelResponse.data.items || channelResponse.data.items.length === 0) {
+        const noChannelError = YouTubeErrorBuilder.createToolError(
+          "No YouTube channel found for this account. Please ensure you have a YouTube channel."
+        );
+        return noChannelError;
+      }
+
+      const channelId = channelResponse.data.items[0].id;
+      if (!channelId) {
+        const invalidChannelError = YouTubeErrorBuilder.createToolError(
+          "Unable to retrieve channel ID. Please ensure your YouTube account is properly set up."
+        );
+        return invalidChannelError;
+      }
+
+      // Create YouTube Analytics API client
+      const ytAnalytics = createYouTubeAnalyticsClient(
+        tokenValidation.tokens!.access_token,
+        tokenValidation.tokens!.refresh_token ?? undefined
+      );
+
+      // Query estimated revenue for the specified date range
+      const response = await ytAnalytics.reports.query({
+        ids: `channel==${channelId}`,
+        startDate: finalStartDate,
+        endDate: finalEndDate,
+        metrics: "estimatedRevenue",
+        dimensions: "day",
+        sort: "day",
+      });
+
+      // Process the response
+      const rows = response.data.rows || [];
+      
+      if (rows.length === 0) {
+        // No revenue data - channel might not be monetized
+        const noRevenueData = YouTubeErrorBuilder.createToolError(
+          "No revenue data found. This could mean your channel is not monetized or you don't have the required Analytics scope permissions. Please ensure your channel is eligible for monetization and you've granted Analytics permissions."
+        );
+        return noRevenueData;
+      }
+
+      // Parse daily revenue data
+      const dailyRevenue = rows.map((row: (string | number)[]) => ({
+        date: String(row[0]), // Day dimension (YYYY-MM-DD format)
+        revenue: parseFloat(String(row[1])) || 0 // Estimated revenue
+      }));
+
+      // Calculate total revenue
+      const totalRevenue = dailyRevenue.reduce((sum, day) => sum + day.revenue, 0);
+
+      const returnResult = YouTubeErrorBuilder.createToolSuccess(
+        `YouTube revenue data retrieved successfully for ${dailyRevenue.length} days. Total revenue: $${totalRevenue.toFixed(2)}`,
+        {
+          revenueData: {
+            totalRevenue: parseFloat(totalRevenue.toFixed(2)),
+            dailyRevenue,
+            dateRange: {
+              startDate: finalStartDate,
+              endDate: finalEndDate,
+            },
+            channelId,
+            isMonetized: true,
+          }
+        }
+      );
+      return returnResult;
+
+    } catch (error: unknown) {
+      console.error("YouTube revenue tool unexpected error:", error);
```

Comment:
your catch is nearly 40 lines of code.
- required: catch is ~2 lines of code
- why is so much code required in this catch?
- what would be lost if you removed it all?